readthedocs/projects/tasks/builds.py

```diff
@@ -513,48 +513,88 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         # Update build object
         self.data.build['success'] = False
 
+    def get_valid_artifact_types(self):
+        """
+        Return a list of all the valid artifact types for this build.
+
+        It performs the following checks on each output format type path:
+         - it exists
+         - it is a directory
+         - does not contains more than 1 files (only PDF, HTMLZip, ePUB)
+
+        TODO: remove the limitation of only 1 file.
+        Add support for multiple PDF files in the output directory and
+        grab them by using glob syntaxt between other files that could be garbage.
+        """
+        for artifact_type in ARTIFACT_TYPES:
+            artifact_directory = self.data.project.artifact_path(
+                version=self.data.version.slug,
+                type_=artifact_type,
+            )
+            if not os.path.exists(artifact_directory):
+                # There is no output directory.
+                # Skip this format.
+                continue
+
+            if not os.path.isdir(artifact_directory):
```

Comment:
This is the correct logic that I had issue with here: https://github.com/readthedocs/readthedocs.org/pull/9939/files#r1087009013

Reply:
He, yeah, I made a mistake in the other PR and it seems I fixed it here 😅